content/blog/nanoparquet-0-3-0/index.Rmd

```diff
@@ -0,0 +1,341 @@
+```{r}
+library(nycflights13)
+flights
+```
+
+First we check how columns of `flights` will be mapped to Parquet types:
+
+```{r}
+parquet_column_types(flights)
+```
+
+This looks fine, so we go ahead and write out the file. By default it will
+be Snappy-compressed, and many columns will be dictionary encoded.
+
+```{r}
+write_parquet(flights, "flights.parquet")
+```
+
+### Parquet metadata
+
+Use `parquet_schema()` to see the schema of a Parquet file. The schema
+also includes "internal" parquet columns. Every Parquet file is a tree
+where columns may be part of an "internal" column. nanoparquet currently
+only supports flat files, that consist of a single internal root column
+and all other columns are leaf columns and are children of the root:
+
+```{r}
+parquet_schema("flights.parquet")
+```
+
+To see more information about a Parquet file, use `parquet_metadata()`:
+
+```{r}
+parquet_metadata("flights.parquet")
+```
+
+The output will include the schema, as above, but also data about the
+row groups (`write_parquet()` always writes a single row group currently),
+and column chunks. There is one column chunk per column in each row group.
+
+The columns chunk information also tells you whether a column chunk is
+dictionary encoded, its encoding, its size, etc.
+
+```{r}
+cc <- parquet_metadata("flights.parquet")$column_chunks
+cc[, c("column", "encodings", "dictionary_page_offset")]
+```
+
+```{r}
+cc[["encodings"]][1:3]
+```
+
+## Limitations
+
+nanoparquet 0.3.0 has a number of limitations.
+
+* **Only flat tables**. `read_parquet()` can only read flat tables, i.e.
+  Parquet files where all leaf columns are children of a single root
+  column. Similarly, `write_parquet()` will not write list columns.
+
+* **Unsupported Parquet types**. `read_parquet()` reads some Parquet types
+  as raw vectors of a list column currently, e.g. `FLOAT16`, `INTERVAL`.
+  See [the manual](https://r-lib.github.io/nanoparquet/reference/nanoparquet-types.html)
+  for details.
+
+* **No encryption**. Encrypted Parquet files are not supported.
+
+* **Missing compression codecs**. `LZO`, `BROTLI` and `LZ4` compression is
+  not yet supported.
+
+* **No statistics**. nanoparquet does not read or write statistics, e.g.
+  minimum and maximum values from and to Parquet files.
+
+* **No checksums**. nanoparquet does not check or write checksums
+  currently.
+
+* **No Bloom filters**. nanoparquet does not currently support reading
+  or writing Bloom filters from or to Parquet files.
+
+* **May be slow for large files**. Being single-threaded and not fully
+  optimized, nanoparquet is probably not suited well for large data sets.
+  It should be fine for a couple of gigabytes. It may be fine if all the
+  data fits into memory comfortably.
+
+* **Single row group**. `write_parquet()` always creates a single row
+  group, which is not optimal for large files.
+
+* **Automatic encoding**. It is currently not possible to choose encodings
+  in `write_parquet()` manually.
+
```

Comment:
I think it's worth a note here that we'll try to tackle (some) of these over time, while continuing to keep nanoparquet lightwight. And encourage folks to submit issues if there are features that we don't support that are important for them.